src/cryptoadvance/specter/controller.py

```diff
@@ -351,8 +351,14 @@ def general_settings():
                 as_attachment=True
             )
         elif action == "restore":
-            restore_devices = json.loads(request.form.get("restoredevices", "[]"))
-            restore_wallets = json.loads(request.form.get("restorewallets", "[]"))
+            if request.form['restoredevices'] != "":
+                restore_devices = json.loads(request.form['restoredevices'])
+            else:
+                restore_devices = []
+            if request.form['restorewallets'] != "":
+                restore_wallets = json.loads(request.form['restorewallets'])
+            else:
+                restore_wallets = []
```

Comment:
I think this is a bit cleaner if you don't mind:
```suggestion
            restore_devices = []
            restore_wallets = []
            if request.form.get("restoredevices", ""):
                restore_devices = json.loads(request.form.get("restoredevices", "[]"))
            if request.form.get("restorewallets", ""):
                restore_wallets = json.loads(request.form.get("restorewallets", "[]"))
```